elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -90,64 +99,76 @@ private void splitFilters() {
-                } catch (IllegalArgumentException e) {
-                    throw new InvalidOperationException(String.format("Invalid grain %s", requestedGrainName));
-                }
+                        //The first grain is the default.
+                        requestedGrainDefinition = timeDim.getSupportedGrains().stream()
+                                .findFirst()
+                                .orElseThrow(() -> new IllegalStateException(
+                                        String.format("Requested default grain, no grain defined on %s",
+                                                attribute.getName())));
+                    } else {
+                        String requestedGrainName = timeGrainArgument.getValue().toString();
 
-                requestedGrainDefinition = timeDim.getSupportedGrains().stream()
-                        .filter(supportedGrainDef -> supportedGrainDef.grain().equals(requestedGrain))
-                        .findAny()
-                        .orElseThrow(() -> new InvalidOperationException(
-                                String.format("Requested grain %s, not supported on %s",
-                                        requestedGrainName, attribute.getName())));
-            }
+                        TimeGrain requestedGrain;
+                        try {
+                            requestedGrain = TimeGrain.valueOf(requestedGrainName);
+                        } catch (IllegalArgumentException e) {
+                            throw new InvalidOperationException(String.format("Invalid grain %s", requestedGrainName));
+                        }
 
-            timeDims.add(timeDim.toProjectedDimension(requestedGrainDefinition));
-        }));
+                        requestedGrainDefinition = timeDim.getSupportedGrains().stream()
+                                .filter(supportedGrainDef -> supportedGrainDef.grain().equals(requestedGrain))
+                                .findAny()
+                                .orElseThrow(() -> new InvalidOperationException(
+                                        String.format("Requested grain %s, not supported on %s",
+                                                requestedGrainName, attribute.getName())));
+                    }
 
-        return timeDims;
+                    return timeDim.toProjectedDimension(requestedGrainDefinition);
+                })
+                .collect(Collectors.toCollection(LinkedHashSet::new));
     }
 
     /**
-     * Gets dimensions based on relationships and attributes from {@link EntityProjection}.
+     * Gets dimensions except time dimensions based on relationships and attributes from {@link EntityProjection}.
      */
     private Set<DimensionProjection> resolveNonTimeDimensions() {
 
         Set<String> allColumns = getAttributes();
         allColumns.addAll(getRelationships());
+        Set<String> sortingFields = getSortingFields(entityProjection.getSorting());
```

Comment:
We shouldn't need to add the sortingFields.  If we are sorting by the field, we are also projecting by it (and the query validation logic can make sure that's the case).  Let's remove this redundancy.